ProgrammaticUI/Controllers/HomeViewController.swift

```diff
@@ -2,12 +2,13 @@ import UIKit
 
 class HomeViewController: UIViewController {
 
-    let stackView = UIStackView()
-    private var kmsLabel: UILabel!
-    private var caloriesLabel: UILabel!
-    private var durationLabel: UILabel!
-    let imageFitness = SAImageView(imageNamed: "fitness.png")
-    var insertData = InsertDataViewController()
+    private let stackView = UIStackView()
+    private let summaryLabel = SALabel(texto: "Summary", font: .boldSystemFont(ofSize: 40))
+    private var kmsLabel = SALabel(texto: "Kms: 0.0", font: .systemFont(ofSize: 20))
+    private var caloriesLabel = SALabel(texto: "Calories: 0", font: .systemFont(ofSize: 20))
+    private var durationLabel = SALabel(texto: "Duration: 00:00:00", font: .systemFont(ofSize: 20))
+    private let imageFitness = SAImageView(imageNamed: "fitness.png")
+    private var insertData = InsertDataViewController()
```

Comment:
`insertData` is only used later on in loc 17 to set its delegate but it is never presented nor used at all. 
```suggestion
```

Note: Strongly retaining a VC should pop an alarm. While doing things like this one needs to be specially careful to avoid cycle retentions.